### ISSUE TITLE
Do you support locales with countries?

### ISSUE THREAD

**Issue by ptarjan** [twitter/twitter-cldr-rb#186]
For example:

```
[1] pry(main)> 500.localize(:"fr-FR").to_currency().to_s(:currency => "USD")
=> "$500.00"
[2] pry(main)> 500.localize(:"fr-CA").to_currency().to_s(:currency => "USD")
=> "500,00 $"
[3] pry(main)> 500.localize(:fr).to_currency().to_s(:currency => "USD")
=> "500,00 $"
```

What is going on with `fr-FR`? That doesn't seem to be how french write currency numbers.

**Comment by camertron** [comment]
Hey @ptarjan, what you're seeing here is unfortunately a current limitation of the library. We only support perhaps 1/5 of all the languages contained in the CLDR data set. This includes fr and fr-CA, but not fr-FR. The behavior you're seeing with fr-FR happens because TwitterCLDR doesn't have any data for that locale and falls back to English.

#### Solution 1: Choose a Different Locale

There are a couple of useful tools you can use to determine what's supported. First, use the `supported_locale?` method to determine if the library has support for any given locale:

```ruby
TwitterCldr.supported_locale?(:fr)  # => true
TwitterCldr.supported_locale?(:"fr-CA")  # => true
TwitterCldr.supported_locale?(:"fr-FR")  # => false :(
```

Second, you can make use of the "likely subtags" feature to pare down your locale to something the library _does_ support:

```ruby
# parse and normalize locale, get maximal subtags
max_locale = TwitterCldr::Shared::Locale.parse('fr-FR').maximize

# get list of locale candidates
candidates = [
  max_locale.dasherized, "#{max_locale.language}-#{max_locale.region}", "#{max_locale.language}"
]

# find supported locale
locale = candidates.find { |c| TwitterCldr.supported_locale?(c) }
```
Admittedly, this is something the library can and should do for you. It's on our road map, but we'd love a pull request if you've got the time to dig in.

#### Solution 2: Add the Missing Locale

We're open to adding any missing locales you want, provided they're supported by CLDR. It should simply be a matter of adding a directory to `resources/locales`, then running `bundle exec rake update`.

**Comment by ptarjan** [comment]
Thanks for the response. I guess falling back to `:en` was the surprising thing. I would have expected it to raise, but maybe that isn't Ruby-ish.

I don't see `TwitterCldr::Shared::Locale` in my version (3.2.1). I'll do update and see if that helps.

I like solution 2. Is there a mechanical transform we can do and just import the whole CLDR? Size of gem probably isn't an issue since it is a server-side component, and completeness would be nice.

**Comment by camertron** [comment]
I'd say that falling back to some kind of supported language is the way to go, and if none can be found then raise. Could also print a warning message saying the exact locale couldn't be found.

`Locale` should have been released with v3.3.0.

We can theoretically import the entire CLDR, but I'm a hesitant to do that. Supporting a locale means supporting all the library's features for that locale, including the complicated stuff (eg. rule-based number formatting, collation, etc). It also means dramatically increasing the size and running time of our test suite. So I'm fine importing individual languages or language groups, but importing all locales wholesale isn't something I'd want to wrestle with right now.

**Comment by camertron** [comment]
A couple of things to mention here:

1. Adding locales can now be done via a rake task, eg `bundle exec rake add_locale[fr-FR]`. The task should be run once under MRI and again under a recent version of JRuby. You can also run the `script/add_locale.sh <locale>` script, which will invoke the rake tasks using the correct ruby versions.
2. A method now exists on `Locale` to find the most specific supported locale: `TwitterCldr::Shared"::Locale.parse('fr-FR').max_supported`,
